src/validator/Validator.cpp

```diff
@@ -309,7 +309,7 @@ StatusOr<Value::Type> Validator::deduceExprType(const Expression* expr) const {
 }
 
 Status Validator::deduceProps(const Expression* expr, ExpressionProps& exprProps) {
-    DeducePropsVisitor visitor(qctx_, space_.id, &exprProps);
+    DeducePropsVisitor visitor(qctx_, space_.id, &exprProps, &userDefinedVarNameList_);
```

Comment:
This implies that we must check duplicate columns after props deduction. I think it's not proper to ensure the order in each subclass of validator.